### PR TITLE
Improved error logging in cases where container is termianted

### PR DESCRIPTION
**Pull request by morsecodist**
Small quality of life improvement just because. Container termination error logging has caused endless confusion for devs.

When the container is terminated, in practice always due to too much memory usage or a timeout, we get no error logged except for the error generated by this check:

```bash
if jq -re .error $OF; then
```

We are checking if the output file exits before uploading it, which works, but the issue is if it is missing a file not found error is logged to stderr. This means when a container is terminated the only error is a file not found error and devs start debugging there.

I piped the stderr of that check to `/dev/null` to avoid confusion, we will never need that error output it is just used for a check. Also, though this isn't true in all cases, in practice with batch we get hit with a `SIGKILL` (exit code 137) for using too much memory and a `SIGTERM` (exit code 143) for timeouts. To make things easier on devs I am adding messages that communicate the likely sources of these signals so they have a better spot to start debugging.

I am also disabling localstack tests here. The moto tests should provide the same degree of coverage. I created an issue to fix the localstack tests but for now I want to get back to only merging PRs that pass and I am comfortable relying on moto. See fix issue: https://github.com/chanzuckerberg/swipe/issues/92

### CODE REVIEW

File: scripts/init.sh [hunk 1]
```diff
@@ -70,9 +70,21 @@ $aws s3 cp "$WDL_WORKFLOW_URI" .
...
 handle_error() {
+  # Add enhanced logging for our most common termination types
+  EXIT_CODE=$?
+  if [[ $EXIT_CODE == 137 ]]; then
+    echo ERROR: container terminated with SIGKILL, this is most likely because memory usage was above container limits >> /dev/stderr
+    exit $EXIT_CODE
+  fi
+
+  if [[ $EXIT_CODE == 143 ]]; then
+    echo ERROR: container terminated with SIGTERM, this is most likely because of a timeout >> /dev/stderr
```
[the remark this answers is not in the document]
Comment (morsecodist) [reply scripts/init.sh:81]: Good point, we do print warnings in that case so we can tell the difference but I will add that to the description so they don't contradict each other

File: scripts/init.sh [hunk 2]
```diff
@@ -70,9 +70,21 @@ $aws s3 cp "$WDL_WORKFLOW_URI" .
...
+
+  if [[ $EXIT_CODE == 143 ]]; then
+    echo ERROR: container terminated with SIGTERM, this is most likely because of a spot instance termination or a timeout >> /dev/stderr
+    exit $EXIT_CODE
+  fi
+
   OF=wdl_output.json;
   EP=.cause.stderr_file;
-  if jq -re .error $OF; then
+  if jq -re .error $OF 2> /dev/null; then
```
Comment (rzlim08) [inline scripts/init.sh:87]: Would we expect there to be errors thrown here given the above code?

**Review by rzlim08 — APPROVED** [review]
Looks good, just a small question

**Review by jgadling — APPROVED** [review]

File: scripts/init.sh (hunk 2 above)
Comment (morsecodist) [reply scripts/init.sh:87]: Good point, there are theoretically other sources of failure, though unlikely, so I think it is best to leave it

**Pull request merged**